pkg/controller/csi_handler_test.go

```diff
@@ -1087,16 +1120,28 @@ func TestCSIHandler(t *testing.T) {
 			},
 			expectedActions: []core.Action{
 				core.NewPatchSubresourceAction(vaGroupResourceVersion, metav1.NamespaceNone, testPVName+"-"+testNodeName,
-					types.MergePatchType, patch(deleted(va(true, fin, ann)),
-						deleted(vaWithDetachError(va(true, fin, ann), "persistentvolume \"pv1\" not found"))),
+					types.MergePatchType, patch(deleted(va(true, "", ann)),
+						deleted(vaWithDetachError(va(true, "", ann), "persistentvolume \"pv1\" not found"))),
 					"status"),
 				core.NewPatchSubresourceAction(vaGroupResourceVersion, metav1.NamespaceNone, testPVName+"-"+testNodeName,
-					types.MergePatchType, patch(deleted(va(true, fin, ann)),
-						deleted(vaWithDetachError(va(true, fin, ann), "persistentvolume \"pv1\" not found"))),
+					types.MergePatchType, patch(deleted(va(false, "", ann)),
+						deleted(vaWithDetachError(va(false, "", ann), "persistentvolume \"pv1\" not found"))),
 					"status"),
 				core.NewPatchSubresourceAction(vaGroupResourceVersion, metav1.NamespaceNone, testPVName+"-"+testNodeName,
-					types.MergePatchType, patch(deleted(va(true, fin, ann)),
-						deleted(vaWithDetachError(va(true, fin, ann), "persistentvolume \"pv1\" not found"))),
+					types.MergePatchType, patch(deleted(va(false, "", ann)),
+						deleted(vaWithDetachError(va(false, "", ann), "persistentvolume \"pv1\" not found"))),
+					"status"),
+				core.NewPatchSubresourceAction(vaGroupResourceVersion, metav1.NamespaceNone, testPVName+"-"+testNodeName,
```

Comment:
This is odd. This test case is supposed to fail the call the first two times, and then succeed on the 3rd.

Reply:
Weird indeed. The test passes with >= 1 actions and only fails when the first action is removed. I'm trying to puzzle out what's going on.

Reply:
Ok, I figured out what was going on with the test. Turns out it wasn't actually testing what it said it was (wrong verb in the reactor), and even then the expected actions weren't behaving as they probably were expected to. I've clarified it all in a comment (or tried to).

Reply:
Ok, the test case itself is kind of strange, but I can see the benefit for adding an additional call to verify that we do retry.

My one last nit is that there's actually 5 loops but the comment says 4.

Reply:
d'oh :P